src/Microsoft.ML.Transforms/CategoricalHashTransform.cs

```diff
@@ -240,10 +240,16 @@ public sealed class ColumnInfo
         /// <param name="env">Host Environment.</param>
         /// <param name="inputColumn">Name of the input column.</param>
         /// <param name="outputColumn">Name of the output column. If this is null '<paramref name="inputColumn"/>' will be used.</param>
+        /// <param name="hashBits">Number of bits to hash into. Must be between 1 and 30, inclusive.</param>
+        /// <param name="invertHash">Limit the number of keys used to generate the slot name to this many. 0 means no invert hashing, -1 means no limit.</param>
         /// <param name="outputKind">The type of output expected.</param>
-        public OneHotHashEncodingEstimator(IHostEnvironment env, string inputColumn,
-            string outputColumn = null, CategoricalTransform.OutputKind outputKind = Defaults.OutputKind)
-            : this(env, new ColumnInfo(inputColumn, outputColumn ?? inputColumn, outputKind))
+        public OneHotHashEncodingEstimator(IHostEnvironment env,
+            string inputColumn,
+            string outputColumn,
+            int hashBits,
```

Comment:
>hashBits [](start = 16, length = 8)

could we have defaults here?

Reply:
here and `invertHash`

---
In reply to: [231604796](https://github.com/dotnet/machinelearning/pull/1564#discussion_r231604796) [](ancestors = 231604796)